content/_changelogs/10.0.0.md

```diff
@@ -0,0 +1,345 @@
+in more detail and how to change your code to migrate to Cypress 10.0.**
+
+- We introduced several breaking changes to the Cypress configuration both the
+  how the configuration is provided and to the configuration options themselves.
+  For the a detailed explanation of these changes, please consult our
+  [Migration Guide](/guides/references/migration-guide) amd use the 10.0
+  migration workflow by running `cypress open` and selecting the project you
+  would like to migrate. Breaking changes include:
+
+  - Cypress Configuration File
+    - Using a `cypress.json` configuration file is no longer valid and must be
+      replaced with either a `cypress.config.js`, `cypress.config.ts`,
+      `cypress.config.cjs` or `cypress.config.mjs` file. This change provide
+      better flexibility and extensibility of configuration. With this change,
+      If both a `cypress.json` file and `cypress.config.*` file is found or if
+      the CLI flag of `-C,--config-file=*.json` is passed, an error will be
+      thrown. Addressed in
+      [#18221](https://github.com/cypress-io/cypress/pull/18221),
+      [#18300](https://github.com/cypress-io/cypress/pull/18300) and
+      [#20643](https://github.com/cypress-io/cypress/pull/20643).
+    - A Cypress configuration file is now required and passing the CLI flag
+      `-C,--config-file=false` will result in an error, where as previously
+      Cypress accepted `-C,--config-file=false` to fallback to the Cypress
+      default configuration values. By default Cypress will automatically load
+      the config file that matches `cypress.config.{cjs,js,mjs,ts}` if the
+      `-C,--config-file` CLI flag is not passed. Addressed in
+      [#20643](https://github.com/cypress-io/cypress/pull/20643).
+    - A default configuration file will no longer be auto-generated by Cypress
+    if a configuration file does not exist. Instead, Cypress will take you
+    through a Project Setup workflow in `open` mode. This allows you to
+    customize the configuration setup by testing-type (E2E or Component Testing)
+    and for the language and dependencies leveraged in your project.
+    <!-- - [RECOMMENDATION] In 9.4.0 we added Intelligent Code Completion via the `defineConfig()` helper function. We encourage leveraging this helper while migrating your configuration. -->
+
+- Updates have been made to allow users to add `cy.hover()` and `cy.mount()`
+  customs commands using `Cypress.Commands.add()` since we suggested adding a
+  custom command since we do not provide a default implementation. Previously
+  users had to use `Cypress.Command.override()` to provide the custom command.
+  Addresses [#18941](https://github.com/cypress-io/cypress/issues/18941).
+
+**Deprecations:**
+
+**Features:**
+
+- Added support for `cypress.config.cjs` and `cypress.config.mjs` configuration
+  files. Addressed in
+  [#20643](https://github.com/cypress-io/cypress/pull/20643).
```

Comment:
I know cypress.config.js support is technically in prior to 10.0, but since this was never announced this would be added to the changelog. I get that this is technically a feature, but it feels a bit repetitive having just read the changes in the breaking change section.